.devcontainer/Dockerfile

```diff
@@ -120,14 +120,21 @@ RUN unzip -o ubuntu-latest-bin.zip -d /usr/local/bin && rm -rf ubuntu-latest-bin
 
 RUN chmod a+x /usr/local/bin/*
 
-RUN adduser --gecos '' --disabled-password cryptol \
-    && echo "cryptol ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/nopasswd \
-    && chown -R cryptol:cryptol /home/cryptol
-USER cryptol
-
-ENV LANG C.UTF-8
-RUN sudo chsh -s /bin/bash $(whoami)
+# create local non-root user, but mimic host ids to avoid file permission problems
+ARG USER_NAME="cryptol"
+ARG USER_GNAME="cryptol"
+# Copy the .env file with USER_UID and USER_GID set in devcontainer.json
+COPY .env /tmp/.env
+RUN . /tmp/.env && rm /tmp/.env \
+    && groupadd -g ${USER_GID} -o ${USER_GNAME} \
+    && useradd -m -u ${USER_UID} -g ${USER_GID} -o -s /bin/bash ${USER_NAME} \
+    && echo "${USER_NAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/nopasswd \
```

Comment:
Does a Cryptol course user (a student creating and editing files in a mounted `cryptol-course` folder and running Cryptol, SAW, Python, etc., or an author modifying and testing course content with these same tools) need elevated privileges?  Let's leave this line out for now and reconsider if compelling needs arise.

Reply:
As a student on macOS, I was unable to create or edit files in the mounted `cryptol-course` because the UID and GID of the container's `cryptol` did not match the host user that owns the files.  It made use of the course difficult. 
 All I did was change the container user's UID:GID from 1000:1000 to the host user by passing them in cleanly.

The alternative I considered was to create a fake user on the host with 1000:1000 and add those perms manually to the `cryptol-course` files before starting the container.  This just felt like a hack and not something turnkey I wanted for ease of use for students.  I wanted a clean approach that would work for any container.  docker provides a way to pass in UID and GID, but VSCode calls docker directly and I saw no easy way to make use of that.

Reply:
I can remove this line, but it means I will have to maintain the fork because students cannot edit or create files in the course directory when VSCode is connected to the Dev Container.